qualtran/bloqs/chemistry/quad_fermion/givens_bloq.py

```diff
@@ -0,0 +1,218 @@
+
+Following notation from Reference [1] we note that a single 
+ladder operator can be similarity transformed by a basis rotation to produce a linear 
+combination of ladder operators
+$$
+U(Q)a_{q}U(Q)^{\dagger} = \sum_{p}Q_{pq}^{*}a_{p} = \overrightarrow{a}_{q}\\
+U(Q)a_{q}^{\dagger}U(Q)^{\dagger} = \sum_{p}Q_{pq}a_{p}^{\dagger} = \overrightarrow{a}_{q}^{\dagger}
+$$
+Each vector of operators can be implemented by a $N$ (size of basis) Givens rotation unitaries as
+$$
+V_{\overrightarrow{Q}_{q}} a_{0} V_{\overrightarrow{Q}_{q}}^{\dagger} = \overrightarrow{a}_{q} \\
+V_{\overrightarrow{Q}_{q}} a_{0}^{\dagger} V_{\overrightarrow{Q}_{q}}^{\dagger} = \overrightarrow{a}_{q}^{\dagger}
+$$
+where 
+$$
+V_{\overrightarrow{Q}_{q}} = V_{n-1,n-2}(0, \phi_{n-1}) V_{n-2, n-3}(\theta_{n-2}, \phi_{n-2})V_{n-3,n-4}(\theta_{n-2}, \phi_{n-2})...V_{2, 1}(\theta_{1}, \phi_{1})V_{1, 0}(\theta_{0}, \phi_{0})
+$$
+with each $V_{ij}(\theta, \phi) = \mathrm{RZ}_{j}(\pi)\mathrm{R}_{ij}(\theta)$. 
+and $1$ Rz rotation for real valued $\overrightarrow{Q}$.
+
+
+References:
+  1.  Vera von Burg, Guang Hao Low, Thomas H ̈aner, Damian S. Steiger, Markus Reiher, Martin Roetteler, 
+  and Matthias Troyer, “Quantum computing enhanced computational catalysis,” Phys. Rev. Res. 3, 033055 (2021).
+
+"""
+from functools import cached_property
+from typing import Dict
+
+from attrs import frozen
+
+from qualtran import Bloq, BloqBuilder, QBit, QFxp, Signature, SoquetT
+from qualtran.bloqs.basic_gates import CNOT, Hadamard, SGate, XGate
+from qualtran.bloqs.rotations.phase_gradient import AddIntoPhaseGrad
+
+
+@frozen
+class RealGivensRotationByPhaseGradient(Bloq):
+    r"""Givens rotation corresponding to a 2-fermion mode transformation generated by
+
+    $$
+        e^{\theta (a_{i}^{\dagger}a_{j} - a_{j}^{\dagger}a_{i})} =  e^{i \theta (YX + XY) / 2}
+    $$
+
+    corresponding to the circuit
+
+        i: ───X───X───S^-1───X───Rz(theta)───X───X───@───────X───S^-1───
+                  │          │               │       │       │
+        j: ───S───@───H──────@───Rz(theta)───@───────X───H───@──────────
+
+    The rotation is performed by addition into a phase state and the fractional binary for
+    $\theta$ is stored in an additional register.
+
+    The Toffoli cost for this block comes from the cost of two rotations by addition into
+    the phase gradient state which which is $2(b_{\mathrm{grad}}-2)$ where $b_{\mathrm{grad}}$
+    is the size of the phasegradient register.
+
+    References:
+        [Compilation of Fault-Tolerant Quantum Heuristics for Combinatorial Optimization](https://arxiv.org/abs/2007.07391).
+        Section II-C: Oracles for phasing by cost function. Appendix A: Addition for controlled rotations
+    Args:
+        phasegrad_bitsize int: size of phase gradient which is also the size of the register
+                               representing the binary fraction of the rotation angle
```

Comment:
nit: the indendation is off here: should just be one indentation level in.

Reply:
really? I thought we had to column align after the variable. That is the case in OpenFermion. What should it be specifically? 

```
    phasegrad_bitsize int: size of phase gradient which is also the size of the register
    representing the binary fraction of the rotation angle
```
?

Reply:
Similar to the Arg docstring here: https://google.github.io/styleguide/pyguide.html#doc-function-raises, except 4-spaces in rather than two, or say: https://github.com/quantumlib/Qualtran/blob/main/qualtran/linalg/lcu_util.py#L46